lib/locations/locations.go

```diff
@@ -104,11 +103,18 @@ var locationTemplates = map[LocationEnum]string{
 	AuditLog:      "${data}/audit-${timestamp}.log",
 	GUIAssets:     "${config}/gui",
 	DefFolder:     "${userHome}/Sync",
-	FailuresFile:  "${data}/failures-unreported.txt",
 }
 
 var locations = make(map[LocationEnum]string)
 
+// Override allows to change only specific, usually user-supplied paths.
+func Override(location LocationEnum, path string) {
+	switch location {
+	case LogFile, GUIAssets:
+		locations[location] = path
+	}
+}
```

Comment:
It seems error prone to make it silently do nothing for everything except those two locations. Why not do it for any `location` value?

Reply:
I redid the whole function to be more aligned with `SetBaseDir()`, removing this restriction.